packages/models/src/store/store.ts

```diff
@@ -0,0 +1,109 @@
+import { Actor, ActorMessage, MessagePayload } from '../actor'
+import { OutPointString, StoreMessage, StorePath } from './interface'
+import { ChainStorage, StorageOffChain } from './chain-storage'
+import { JSONStorage, JSONStorageOffChain } from './json-storage'
+import { NonExistentException, NonStorageInstanceException } from '../exceptions'
+
+type GetState<T> = T extends ChainStorage<infer State> ? State : never
+type UnknownAsNever<T> = unknown extends T ? (0 extends 1 & T ? T : never) : T
```

Comment:
What does `UnknownAsNever` mean

Reply:
If the developer does not transfer the `data` type in `state`. They should not get `data` by `get` function.
```
get(key: OutPointString, paths: ['witness']): UnknownAsNever<GetState<StorageT>['witness']>
const store = new JSONStore<{ data: { a: number } }>()
const witness = store.get('0x1234', ['witness'])
```
The `witness` will be `never`.